src/store/modules/Configuration/FirmwareStore.js

```diff
@@ -0,0 +1,63 @@
+import api from '../../api';
+
+const FirmwareStore = {
+  namespaced: true,
+  state: {
+    firmwareInfo: null,
+    bmcActiveVersion: '--',
+    hostActiveVersion: '--'
+  },
+  getters: {
+    firmwareInfo: state => state.firmwareInfo,
+    bmcActiveVersion: state => state.bmcActiveVersion,
+    hostActiveVersion: state => state.hostActiveVersion
+  },
+  mutations: {
+    setFirmwareInfo: (state, firmwareInfo) =>
+      (state.firmwareInfo = firmwareInfo),
+    setBmcActiveVersion: (state, bmcActiveVersion) =>
+      (state.bmcActiveVersion = bmcActiveVersion),
+    setHostActiveVersion: (state, hostActiveVersion) =>
+      (state.hostActiveVersion = hostActiveVersion)
+  },
+  actions: {
+    getFirmwareInfo({ commit }) {
+      api
+        .get('/xyz/openbmc_project/software/enumerate')
+        .then(response => {
+          const firmwareInfo = response.data.data;
+          const functionalImages =
+            firmwareInfo['/xyz/openbmc_project/software/functional'].endpoints;
+          for (let key in firmwareInfo) {
+            if (firmwareInfo[key].hasOwnProperty('Version')) {
+              // If "Functional" activation status is
+              // functional, else it is "activation"
+              // github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/Software/Activation.interface.yaml
+              let activationStatus = '';
+              if (firmwareInfo[key].Activation) {
+                activationStatus = firmwareInfo[key].Activation.split(
+                  '.'
+                ).pop();
+              }
+              if (functionalImages.includes(key)) {
+                activationStatus = 'Functional';
```

Comment:
This is a bit nit picky, but if the if activationStatus has to be either `Functional` or some Activation status, I think that should be written as an else if.

Also, that comments breaks up the readability a tad (super nit, i know. )

```
// If "Functional" activation status is functional, else it is "activation"
// github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/Software/Activation.interface.yaml
  if (firmwareInfo[key].hasOwnProperty('Version')) {
    let activationStatus = '';
    if (firmwareInfo[key].Activation) {
      activationStatus = firmwareInfo[key].Activation.split(
         '.'
        ).pop();
    } else if (functionalImages.includes(key)) {
      activationStatus = 'Functional';
    }
```

Reply:
That doesn't work. I don't think the two blocks are dependent on each other